### PR TITLE
Crosswalk between projects and uploads in audit report

### PR DESCRIPTION
**Pull request by ajhyndman**
We probably want to do more with this tab eventually, but mapping between projects and uploads has been highly requested, so we solve that now.

Fixes #340

### CODE REVIEW

File: src/server/services/persist-upload.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ async function persistUpload ({ filename, user, buffer }) {
   return upload
 }
 
-function bufferForUpload (upload) {
+async function bufferForUpload (upload) {
```
Comment (igor47) [inline src/server/services/persist-upload.js:53]: just curious, why is this necessary?

File: src/server/lib/audit-report.js [hunk 2]
```diff
@@ -42,77 +50,120 @@ async function createObligationSheet (periodId) {
...
+    switch (record.type) {
+      case 'ec1':
+      case 'ec2':
+      case 'ec3':
+      case 'ec4':
+      case 'ec5':
+      case 'ec7':
+        rows.push({
+          'Project ID': record.content.Project_Identification_Number__c,
+          'Upload': record.upload.filename // TODO: link to upload in arpa reporter
```
Comment (igor47) [inline src/server/lib/audit-report.js:155]: i'm not sure how useful the filename will be, seems like maybe including at least an upload id column would be convenient, a link would be even better.

File: src/server/services/records.js [hunk 3]
```diff
@@ -36,8 +37,8 @@ function extractRecords (buffer) {
   const subcategory = cover['Detailed Expenditure Category']
 
   const records = [
-    { type: 'certification', content: certification },
-    { type: 'cover', content: cover }
+    { type: 'certification', upload, content: certification },
+    { type: 'cover', upload, content: cover }
```
Comment (igor47) [inline src/server/services/records.js:41]: hmm, do these types of record not have a `subcategory` field?

**Review by igor47 — APPROVED** [review]

File: src/server/services/persist-upload.js (hunk 1 above)
Comment (ajhyndman) [reply src/server/services/persist-upload.js:53]: I don't think it is, actually.

I was fiddling with things because I noticed that bufferForUpload was running in series instead of in parallel when called from `recordsForReportingPeriod`.  I'm still not 100% sure why, but inlining `extractRecords` to `recordsForUpload` seemed to do the trick.

I guess I left it in because otherwise there's nothing indicating that this function is async without reading more code.

File: src/server/lib/audit-report.js (hunk 2 above)
Comment (ajhyndman) [reply src/server/lib/audit-report.js:155]: Yeah, I'm imagining linking the filename to the upload in the tool.  👍 

I'll fix that as soon as we decide on a solution for links.

File: src/server/services/records.js (hunk 3 above)
Comment (ajhyndman) [reply src/server/services/records.js:41]: Hmm, that's true.  I'm not sure we actually use these records for anything other than extracting the subcategory at the moment.  Maybe we should just omit them?
